app/grandchallenge/workstation_configs/models.py

```diff
@@ -264,15 +264,81 @@ def save(self, *args, **kwargs):
 
 class WindowPreset(TitleSlugDescriptionModel):
     width = models.PositiveIntegerField(
-        validators=[MinValueValidator(limit_value=1)]
+        blank=True, null=True, validators=[MinValueValidator(limit_value=1)]
     )
-    center = models.IntegerField()
+    center = models.IntegerField(blank=True, null=True)
+
+    lower_percentile = models.PositiveSmallIntegerField(
+        blank=True, null=True, validators=[MaxValueValidator(limit_value=100)]
+    )
+
+    upper_percentile = models.PositiveSmallIntegerField(
+        blank=True, null=True, validators=[MaxValueValidator(limit_value=100)]
+    )
+
+    def _validate_percentile(self):
+        if self.upper_percentile < self.lower_percentile:
+            raise ValidationError(
+                f"Upper percentile ({self.upper_percentile}%) should be below the "
+                f"lower percentile ({self.lower_percentile}%)"
+            )
+
+    def _validate_fixed(self):
+        pass
+
+    def clean(self):
+        window_center_all = None not in {self.width, self.center}
+        window_center_none = all(v is None for v in {self.width, self.center})
+        percentile_all = None not in {
+            self.lower_percentile,
+            self.upper_percentile,
+        }
+        percentile_none = all(
+            v is None for v in {self.lower_percentile, self.upper_percentile}
+        )
+
+        if window_center_all and percentile_none:
+            self._validate_fixed()
+        elif percentile_all and window_center_none:
+            self._validate_percentile()
+        else:
+            raise ValidationError(
+                "Either (upper and lower percentiles) or (width and center) should be entered"
+            )
 
     class Meta(TitleSlugDescriptionModel.Meta):
         ordering = ("title",)
+        constraints = [
+            models.CheckConstraint(
+                name="%(app_label)s_%(class)s_either_fixed_or_percentile",
+                check=(
+                    models.Q(
+                        center__isnull=False,
+                        width__isnull=False,
+                        lower_percentile__isnull=True,
+                        upper_percentile__isnull=True,
+                    )
+                    | models.Q(
+                        center__isnull=True,
+                        width__isnull=True,
+                        lower_percentile__isnull=False,
+                        upper_percentile__isnull=False,
+                    )
+                ),
+            ),
+            models.CheckConstraint(
+                name="%(app_label)s_%(class)s_upper_gte_lower_percentile",
+                check=models.Q(
+                    upper_percentile__gte=models.F("lower_percentile")
```

Comment:
`_gte` is not what we need, otherwise you definitely end up with a zero width window.

Reply:
I missed that the current `width` has a minimum of `1`! On the CIRRUS side, restrictions needed to be a lot more relaxed in order to account for the `inverse` option so I was in a permissive mindset. Thank you for the spot!